### DISCUSSION TITLE
Add built-in Jupyter kernel like Deno

### DISCUSSION THREAD

**Discussion by nicholasoxford** [oven-sh/bun#8731] [Ideas]
[Deno Jupyter Runtime](https://docs.deno.com/runtime/manual/tools/jupyter)

![image](https://github.com/oven-sh/bun/assets/51415676/4ac734e8-dab8-4679-98b8-8ef80d98ea91)
[https://twitter.com/wesbos/status/1754674921706315949](https://twitter.com/wesbos/status/1754674921706315949)

**Comment by AntonioMeireles** [comment]
@Jarred-Sumner (fwiw) now that bun is, more and more, a drop in node replacement, this (after bun shell release) is (at least for my use cases) the main missing killer feature ;-)

**Reply by btakita** [reply]
With AI & RAG, this would be a great time to prioritize the feature. Drive the entire AI & RAG pipeline from Bun.

See https://www.npmjs.com/package/ijavascript

**Comment by liudonghua123** [comment]
> @Jarred-Sumner (fwiw) now that bun is, more and more, a drop in node replacement, this (after bun shell release) is (at least for my use cases) the main missing killer feature ;-)  

I also like this feature. It would be useful and friendly to js beginners.

**Comment by Artoria2e5** [comment]
A feature-rich notebook like runkit (but local, and with esm!) would be really great for me. I occasionally use a JS REPL for quick jobs (color maps with culori, short proof-of-concepts, etc.), and to put that in a notebook like Python people have would be great.

**Comment by Miciurash** [comment]
@Jarred-Sumner - are there plans for this? is help needed?

**Comment by perezd** [comment]
I would love to see something like this to integrate Bun into Jupyter/VSCode.

**Comment by ntaraujo** [comment]
me too

**Comment by abdelkareemkobo** [comment]
any updates?

**Comment by dbuezas** [comment]
I've been wishing for this for a long while (since tslab stopped working), so I made one for vscode.

It's crazy how fast bun is.

https://marketplace.visualstudio.com/items?itemName=DavidBuezas.bunbook

https://github.com/dbuezas/bunbook

<img width="2084" height="2274" alt="image" src="https://github.com/user-attachments/assets/eb7e6aa7-a260-4bac-8002-d0eeaae0744a" />

**Reply by dbuezas** [reply]
It now has error squiggles, hover types, and variables and functions defined in a cell are visible in the others
